airbyte-ci/connectors/pipelines/pipelines/pipelines/metadata.py

```diff
@@ -2,112 +2,98 @@
-        super().__init__(context, title, METADATA_DIR, METADATA_LIB_MODULE_PATH)
-        self.poetry_run_container = self.poetry_run_container.with_mounted_file(
-            METADATA_FILE_NAME, get_metadata_file_from_path(context, metadata_path)
+class MetadataValidation(SimpleCIStep):
+    def __init__(self, context: ConnectorContext):
+        super().__init__(
+            title=f"Validate metadata for {context.connector.technical_name}",
+            context=context,
+            paths_to_mount=[
+                MountPath(context.connector.metadata_file_path),
+                MountPath(context.connector.documentation_file_path),
+                MountPath(context.connector.icon_path, optional=True),
+            ],
+            internal_tools=[
+                MountPath(METADATA_SERVICE_TOOL_MODULE_PATH),
+            ],
+            command=[
+                "metadata_service",
+                "validate",
+                str(context.connector.metadata_file_path),
+                str(context.connector.documentation_file_path),
+            ],
         )
 
-    async def _run(self) -> StepResult:
-        return await super()._run(["metadata_service", "validate", METADATA_FILE_NAME])
 
-
-class MetadataUpload(PoetryRun):
+class MetadataUpload(SimpleCIStep):
     # When the metadata service exits with this code, it means the metadata is valid but the upload was skipped because the metadata is already uploaded
     skipped_exit_code = 5
 
     def __init__(
         self,
-        context: PipelineContext,
-        metadata_path: Path,
+        context: ConnectorContext,
         metadata_bucket_name: str,
         metadata_service_gcs_credentials_secret: dagger.Secret,
         docker_hub_username_secret: dagger.Secret,
         docker_hub_password_secret: dagger.Secret,
         pre_release: bool = False,
         pre_release_tag: Optional[str] = None,
     ):
-        title = f"Upload {metadata_path}"
-        self.gcs_bucket_name = metadata_bucket_name
-        self.pre_release = pre_release
-        self.pre_release_tag = pre_release_tag
-        super().__init__(context, title, METADATA_DIR, METADATA_LIB_MODULE_PATH)
-
-        # Ensure the icon file is included in the upload
-        base_container = self.poetry_run_container.with_file(METADATA_FILE_NAME, get_metadata_file_from_path(context, metadata_path))
-        metadata_icon_path = metadata_path.parent / METADATA_ICON_FILE_NAME
-        if metadata_icon_path.exists():
-            base_container = base_container.with_file(
-                METADATA_ICON_FILE_NAME, get_metadata_icon_file_from_path(context, metadata_icon_path)
-            )
+        title = f"Upload metadata for {context.connector.technical_name} v{context.connector.docker_image_tag}"
+        command_to_run = [
+            "metadata_service",
+            "upload",
+            str(context.connector.metadata_file_path),
+            str(context.connector.documentation_file_path),
```

Comment:
@bnchrch what actually sets this `documentation_file_path` when this step is automatically run due to merging a change to a connector which bumps the version?

I expected to see some logic in this PR which constructs the path to the connector doc file but I'm not seeing that, so how does the auto-publish on merge do this?

Reply:
Good question!

So we already have a Connector Class that holds onto paths for
1. connector code directory
2. docs
3. metadata
4. icon

You can see the docs path method here:
(https://github.com/airbytehq/airbyte/blob/bnchrch/ci/metadata-doc-step/airbyte-ci/connectors/connector_ops/connector_ops/utils.py#L218)

The I mounted that path using that method here:
https://github.com/airbytehq/airbyte/blob/bnchrch/ci/metadata-doc-step/airbyte-ci/connectors/pipelines/pipelines/pipelines/metadata.py#L79

and again when constructing  the command here:
https://github.com/airbytehq/airbyte/blob/bnchrch/ci/metadata-doc-step/airbyte-ci/connectors/pipelines/pipelines/pipelines/metadata.py#L79

Reply:
That makes it very clear, thanks for explaining!